tests/testlibrary.c

```diff
@@ -362,6 +362,62 @@ test_list_refs (void)
   g_assert_cmpint (refs->len, ==, 0);
 }
 
+static void
+test_list_remote_refs (void)
+{
+  g_autoptr(FlatpakInstallation) inst = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GPtrArray) refs = NULL;
+  int i;
+
+  inst = flatpak_installation_new_user (NULL, &error);
+  g_assert_no_error (error);
+
+  refs = flatpak_installation_list_remote_refs_sync (inst, repo_name, NULL, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (refs);
+  g_assert_cmpint (refs->len, ==, 2);
+
+
+  for (i = 0; i < refs->len; i++)
+    {
+      FlatpakRemoteRef *remote_ref = g_ptr_array_index (refs, i);
+      FlatpakRef *ref = FLATPAK_REF (remote_ref);
+      GBytes *metadata;
+
+      g_assert (ref != NULL);
+
+      if (strcmp ("org.test.Hello", flatpak_ref_get_name (ref)) == 0)
+        {
+          g_assert_cmpint (flatpak_ref_get_kind (ref), ==, FLATPAK_REF_KIND_APP);
+        }
+      else
+        {
+          g_assert_cmpstr (flatpak_ref_get_name (ref), ==, "org.test.Platform");
+          g_assert_cmpint (flatpak_ref_get_kind (ref), ==, FLATPAK_REF_KIND_RUNTIME);
+        }
+
+      g_assert_cmpstr (flatpak_ref_get_branch (ref), ==, "master");
+      g_assert_cmpstr (flatpak_ref_get_commit (ref), !=, NULL);
+      g_assert_cmpstr (flatpak_ref_get_arch (ref), ==, flatpak_get_default_arch ());
+
+      g_assert_cmpstr (flatpak_remote_ref_get_remote_name (remote_ref), ==, repo_name);
+      g_assert_cmpstr (flatpak_remote_ref_get_eol (remote_ref), ==, NULL);
+      g_assert_cmpstr (flatpak_remote_ref_get_eol_rebase (remote_ref), ==, NULL);
+
+      g_assert_cmpuint (flatpak_remote_ref_get_installed_size (remote_ref), >, 0);
+      g_assert_cmpuint (flatpak_remote_ref_get_download_size (remote_ref), >, 0);
+
+      metadata = flatpak_remote_ref_get_metadata (remote_ref);
+      g_assert (metadata != NULL);
+
+      if (strcmp ("org.test.Hello", flatpak_ref_get_name (ref)) == 0)
```

Comment:
Maybe check `flatpak_ref_get_kind` here

Reply:
It doesn't really matter does it? In the end we check both name and kind for both refs. I perfer to switch the checks on the name because that makes it more obvious what is happening, and it scales to adding more refs. The fact that there is just one of each kind is more of a coincidence.

Reply:
Makes sense